sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs

```diff
@@ -3469,8 +3469,12 @@ public static partial class SearchModelFactory
         public static Azure.Search.Documents.Indexes.Models.SearchIndexStatistics SearchIndexStatistics(long documentCount = (long)0, long storageSize = (long)0, long vectorIndexSize = (long)0) { throw null; }
         public static Azure.Search.Documents.Indexes.Models.SearchResourceCounter SearchResourceCounter(long usage, long? quota) { throw null; }
         public static Azure.Search.Documents.Models.SearchResultsPage<T> SearchResultsPage<T>(Azure.Search.Documents.Models.SearchResults<T> results) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
```

Comment:
Where did these come from? I don't see the changes in the PR. If from before, this means the checks would've failed prior.

Reply:
We have added `[EditorBrowsable(EditorBrowsableState.Never)]` into the [SearchResults.cs](https://github.com/Azure/azure-sdk-for-net/pull/41694/files#diff-5c0e533bf06ba67f92e99afce2b578bb80514e7aa77ca02cc58c59651787ded1) file in this PR and added a new mocking constructor. This was added as the SemanticSearchResults property was missing in the existing mocking constructor.

Reply:
Why are you doing static factory methods on the types themselves, though? Our [guidelines state](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking-factory-builder):

> ✅ DO provide factory or builder for constructing model graphs returned from virtual service methods.

> 
> Model types shouldn’t have public constructors. Instances of the model are typically returned from the client library, and are not constructed by the consumer of the library. Mock implementations need to create instances of model types. Implement a static class called <service>ModelFactory in the same namespace as the model types:

Or are you just continuing a trend this library already seems to have? Has it GA'd this way through architect approval?

Reply:
@heaths - The static factory method is not on the type itself. [SearchModelFactory](https://github.com/Azure/azure-sdk-for-net/blob/588852238962ab0f08f8bc2143ea7e50bfc90577/sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs#L433) is a static partial class created inside the `SearchResults.cs` file, and all the factory methods are inside the [SearchModelFactory](https://github.com/Azure/azure-sdk-for-net/blob/588852238962ab0f08f8bc2143ea7e50bfc90577/sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs#L433-L477) type.

Reply:
I see now. That's really confusing to reviewers and maintainers and requires suppressing a warning not to put multiple types in a single source file. I think we should avoid this pattern in the future.

Reply:
I completely agree with that. I have been following to the existing pattern, but I will create a separate PR for this cleanup.